database-commons/src/main/java/co/cask/CommonSchemaReader.java

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask;
+
+import co.cask.cdap.api.data.schema.Schema;
+import co.cask.cdap.api.data.schema.UnsupportedTypeException;
+import com.google.common.base.Strings;
+import com.google.common.collect.Lists;
+
+import java.io.IOException;
+import java.sql.ResultSet;
+import java.sql.ResultSetMetaData;
+import java.sql.SQLException;
+import java.sql.Types;
+import java.util.List;
+import javax.annotation.Nullable;
+
+
+/**
+ * Common schema reader for mapping non specific DB types.
+ */
+public class CommonSchemaReader implements SchemaReader {
+
+  protected final ResultSet resultSet;
+
+  public CommonSchemaReader(ResultSet resultSet) {
+    this.resultSet = resultSet;
+  }
+
+  public List<Schema.Field> getSchemaFields(@Nullable String schemaStr)
```

Comment:
missed Override annotation everywhere in this class

Reply:
added